kolibri/content/models.py

```diff
@@ -0,0 +1,333 @@
+    Things it can represent are, for example, video, exercise, audio or document...
+    """
+    content_id = models.UUIDField(primary_key=False, default=uuid4, editable=False)
+    title = models.CharField(max_length=200)
+    description = models.CharField(max_length=400, blank=True, null=True)
+    kind = models.CharField(max_length=50)
+    slug = models.CharField(max_length=100)
+    total_file_size = models.IntegerField()
+    available = models.BooleanField(default=False)
+    license = models.ForeignKey('License')
+    prerequisite = models.ManyToManyField('self', related_name='is_prerequisite_of', through='PrerequisiteContentRelationship', symmetrical=False, blank=True)
+    is_related = models.ManyToManyField('self', related_name='relate_to', through='RelatedContentRelationship', symmetrical=False, blank=True)
+    parent = TreeForeignKey('self', null=True, blank=True, related_name='children', db_index=True)
+
+    class Meta:
+        verbose_name = 'Content Metadata'
+
+    class Admin:
+        pass
+
+    def __str__(self):
+        return self.title
+
+class MimeType(AbstractContent):
+    """
+    Normalize the "kind"(mimetype) of Format model
+    """
+    readable_name = models.CharField(max_length=50)
+    machine_name = models.CharField(max_length=100)
+
+    class Admin:
+        pass
+
+    def __str__(self):
+        return self.readable_name
+
+class Format(AbstractContent):
+    """
+    The intermediate layer of the contentDB schema, defines a complete set of resources that is ready to be rendered on the front-end,
+    including the quality of the content.
+    Things it can represent are, for example, high_resolution_video, low_resolution_video, vectorized_video, khan_excercise...
+    """
+    available = models.BooleanField(default=False)
+    format_size = models.IntegerField(blank=True, null=True)
+    quality = models.CharField(max_length=50, blank=True, null=True)
+    contentmetadata = models.ForeignKey(ContentMetadata, related_name='formats', blank=True, null=True)
+    mimetype = models.ForeignKey(MimeType, blank=True, null=True)
+
+    class Admin:
+        pass
+
+class File(AbstractContent):
+    """
+    The bottom layer of the contentDB schema, defines the basic building brick for content.
+    Things it can represent are, for example, mp4, avi, mov, html, css, jpeg, pdf, mp3...
+    """
+    checksum = models.CharField(max_length=400, blank=True, null=True)
+    extension = models.CharField(max_length=100, blank=True, null=True)
+    available = models.BooleanField(default=False)
+    file_size = models.IntegerField(blank=True, null=True)
+    content_copy = models.FileField(upload_to=content_copy_name, storage=ContentCopyStorage(), max_length=200, blank=True)
+    format = models.ForeignKey(Format, related_name='files', blank=True, null=True)
+
+    class Admin:
+        pass
+
+    def __str__(self):
+        return '{checksum}{extension}'.format(checksum=self.checksum, extension=self.extension)
```

Comment:
I only use this __str__ function for human friendly terminal display, mostly when debugging.
The file path is kinda opaque, handled by the model API through FileField